com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/ShaderGraphCommandOverrides.cs

```diff
@@ -231,5 +231,24 @@ internal static void HandlePasteSerializedData(UndoStateComponent undoState, Gra
                 }
             }
         }
+
+        internal static void HandleChangeNodeMode(UndoStateComponent undoState, GraphModelStateComponent graphModelState, ChangeNodeModeCommand command)
+        {
+            using (var undoStateUpdater = undoState.UpdateScope)
+            {
+                undoStateUpdater.SaveState(graphModelState);
+            }
+
+            using (var graphUpdater = graphModelState.UpdateScope)
+            using (var changeScope = graphModelState.GraphModel.ChangeDescriptionScope)
+            {
+                if (command.NodeModel is SGNodeModel sgNodeModel)
+                {
+                    sgNodeModel.ChangeNodeFunction(command.NodeModel.Modes[command.NewNodeModeIndex]);
+                    command.NodeModel.ChangeMode(command.NewNodeModeIndex);
```

Comment:
Maybe `ChangeNodeFunction` could become `override ChangeMode` ?

Reply:
I would need to make `NodeModel.ChangeMode` virtual but that's not an issue (and might be a good idea actually).